src/mscorlib/src/System/ReadOnlySpan.cs

```diff
@@ -73,6 +79,8 @@ public ReadOnlySpan(T[] array, int start, int length)
         {
             if (array == null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            if (default(T) == null && array.GetType() != typeof(T[]))
```

Comment:
I will remove it here, disable the test, and create an issue to address this in corefx.